diff --git a/node_modules/@myriaddreamin/vite-plugin-typst/.bun-tag-aa0f5db47a342345 b/.bun-tag-aa0f5db47a342345
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/index.js b/dist/index.js
index 669e484570cebe11a9beb1f97791996f1208f708..7148883ca0214fc936d32a7f0b3223205b653922 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -80,14 +80,16 @@ export default parts;`;
             }
             return `export default ${JSON.stringify(result.result.html())}`;
         },
-        resolveId(source) {
-            const { path, attributes } = extractOpts(source);
-            if (!path.endsWith('.typ'))
-                return null;
-            if (attributes.html || attributes.parts) {
-                return source + suffixJs;
-            }
-            return provider.resolveRel(path);
+        resolveId(source, importer) {
+          const { path: filepath, attributes } = extractOpts(source);
+          if (!filepath.endsWith(".typ")) return null;
+          const abspath = path.isAbsolute(filepath)
+            ? filepath
+            : path.resolve(path.dirname(importer), source);
+          if (attributes.html || attributes.parts) {
+            return abspath + suffixJs;
+          }
+          return provider.resolveRel(abspath);
         },
         config(conf) {
             viteReload(conf);
